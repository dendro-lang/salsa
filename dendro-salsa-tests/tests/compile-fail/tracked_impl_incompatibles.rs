#[salsa::jar(db = Db)]
struct Jar(MyTracked);


#[salsa::tracked]
struct MyTracked {
    field: u32,
}

#[salsa::tracked(return_ref)]
impl std::default::Default for MyTracked {
    fn default() -> Self {

    }
}
#[salsa::tracked(specify)]
impl std::default::Default for MyTracked {
    fn default() -> Self {

    }
}

#[salsa::tracked(no_eq)]
impl std::default::Default for MyTracked {
    fn default() -> Self {

    }
}

#[salsa::tracked(data = Data)]
impl std::default::Default for MyTracked {
    fn default() -> Self {

    }
}

#[salsa::tracked(db = Db)]
impl std::default::Default for MyTracked {
    fn default() -> Self {

    }
}

#[salsa::tracked(recover_fn = recover)]
impl std::default::Default for MyTracked {
    fn default() -> Self {

    }
}

#[salsa::tracked(lru = 32)]
impl std::default::Default for MyTracked {
    fn default() -> Self {

    }
}

#[salsa::tracked(constructor = Constructor)]
impl std::default::Default for MyTracked {
    fn default() -> Self {

    }
}
#[salsa::tracked]
impl std::default::Default for [MyTracked; 12] {
    fn default() -> Self {

    }
}


trait Db: salsa::DbWithJar<Jar> {}

fn main() {}