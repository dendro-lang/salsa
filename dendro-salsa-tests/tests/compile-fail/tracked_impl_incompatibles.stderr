error: `return_ref` option not allowed here
  --> tests/compile-fail/tracked_impl_incompatibles.rs:10:18
   |
10 | #[salsa::tracked(return_ref)]
   |                  ^^^^^^^^^^

error: `specify` option not allowed here
  --> tests/compile-fail/tracked_impl_incompatibles.rs:16:18
   |
16 | #[salsa::tracked(specify)]
   |                  ^^^^^^^

error: `no_eq` option not allowed here
  --> tests/compile-fail/tracked_impl_incompatibles.rs:23:18
   |
23 | #[salsa::tracked(no_eq)]
   |                  ^^^^^

error: `data` option not allowed here
  --> tests/compile-fail/tracked_impl_incompatibles.rs:30:18
   |
30 | #[salsa::tracked(data = Data)]
   |                  ^^^^

error: `db` option not allowed here
  --> tests/compile-fail/tracked_impl_incompatibles.rs:37:18
   |
37 | #[salsa::tracked(db = Db)]
   |                  ^^

error: unrecognized option `recover_fn`
  --> tests/compile-fail/tracked_impl_incompatibles.rs:44:18
   |
44 | #[salsa::tracked(recover_fn = recover)]
   |                  ^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/tracked_impl_incompatibles.rs:51:18
   |
51 | #[salsa::tracked(lru = 32)]
   |                  ^^^

error: `constructor` option not allowed here
  --> tests/compile-fail/tracked_impl_incompatibles.rs:58:18
   |
58 | #[salsa::tracked(constructor = Constructor)]
   |                  ^^^^^^^^^^^

error: #[salsa::tracked] can only be applied to salsa structs
  --> tests/compile-fail/tracked_impl_incompatibles.rs:65:32
   |
65 | impl std::default::Default for [MyTracked; 12] {
   |                                ^^^^^^^^^^^^^^^
